index.js

```diff
@@ -1 +1,266 @@
+ * @return {cluster~Worker} Newly created worker.
+ */
+ClusterManager.prototype.createWorker = function () {
+  var self = this;
+  var worker = this.cluster.fork();
+
+  // Deals with unhandled worker errors
+  var workerDomain = domain.create();
+  workerDomain.add(worker);
+  workerDomain.on('error', function (err) {
+    self.log.error('Unhandled worker error: ' + err.stack);
+    worker.process.kill(1);
+  });
+
+  this.workers.push(worker);
+  this.log.info('Created new worker: ' + worker.id);
+  return worker;
+};
+
+/**
+ * Handles worker `fork` events. This event is emitted when a worker is forked
+ * off the master cluster.
+ * @param {cluster~Worker} Worker that was forked.
+ */
+ClusterManager.prototype.fork = function (worker) {
+  this.log.info('Worker forked: ' + worker.id);
+};
+
+/**
+ * Handles worker `listening` events. Indicates to the master that a particular
+ * worker is listening.
+ * @param {cluster~Worker} Worker that is now listening.
+ * @param address Address on which the worker is listening.
+ */
+ClusterManager.prototype.listening = function (worker, address) {
+  this.log.info([
+    'Worker listening:', worker.id,
+    'on address', (address.address+':'+address.port)
+  ].join(' '));
+};
+
+/**
+ * Handles worker `exit` events.
+ * @param {cluster~Worker} worker Worker that exited.
+ * @param {Number} code Exit code for the worker process.
+ * @param {String} signal Signal name that caused the process to be killed.
+ */
+ClusterManager.prototype.exit = function (worker, code, signal) {
+  this.log.info([
+    'Worker exited:', worker.id,
+    '-- with status:', code,
+    '-- and signal:', signal
+  ].join(' '));
+
+  var self = this;
+  this.workers.map(pluck('id')).some(function (workerId, i) {
+    if (workerId === worker.id) {
+      self.workers.splice(i, 1);
+    }
+  });
+};
+
+/**
+ * Handles worker `online` events. This indicates to the cluster that a worker
+ * process has successfully spawned a process and is running.
+ * @param {cluster~Worker} worker Worker that came online.
+ */
+ClusterManager.prototype.online = function (worker) {
+  this.log.info('Worker online: ' + worker.id);
+};
+
+/**
+ * Handles worker `disconnect` events. This indicates that the worker has
+ * disconnected from communication but is not nessessarily dead.
+ * @param {cluster~Worker} worker Worker that disconnected.
+ */
+ClusterManager.prototype.disconnect = function (worker) {
+  this.log.info('Worker disconnected: ' + worker.id + ' -- killing');
+};
+
+/**
+ * Called when master process domain encounters an unhandled error. By default
+ * this method will log the error stack, indicate that the error is fatal, and
+ * kill the process with a status code `1`.
+ * @param {Error} err Unhandled error on the master process.
+ */
+ClusterManager.prototype.masterError = function(err) {
+  this.log.error('Unhandled master error: ' + err.stack);
+
+  // TODO Should we really kill the process here by default?
```

Comment:
So you can extend the default behavior by subclassing the `ClusterManager` class and overriding the `masterError` method. If you want to add additional event handlers you can do that directly on the cluster, like so:

```js
manager.cluster.on('error', function myErrorHandler() {
  // Do additional handling...
});
```

I like the idea of having the kill behavior configurable through an option, reduces the need to subclass for minor changes to default functionality.